src/System.Text.Json/tests/Serialization/Stream.ReadTests.cs

```diff
@@ -139,5 +139,98 @@ public static async Task TestBOMWithNoJsonValue()
                     async () => await JsonSerializer.DeserializeAsync<byte>(stream));
             }
         }
+
+        public static IEnumerable<object[]> BOMWithJsonElementTestData
+        {
+            get
+            {
+                yield return CreateStreamArgs(100);
+                yield return CreateStreamArgs(200);
+                yield return CreateStreamArgs(400);
+                yield return CreateStreamArgs(800);
+                yield return CreateStreamArgs(1600);
+
+                static object[] CreateStreamArgs(int count)
+                {
+                    byte[] objBytes = Encoding.UTF8.GetBytes(
+                        @"{""Test"":{},""Test2"":[],""Test3"":{""Value"":{}},""PersonType"":0,""Id"":2}");
+
+                    byte[] utf8Bom = Encoding.UTF8.GetPreamble();
+                    byte[] comma = Encoding.UTF8.GetBytes(@",");
+                    byte[] startArray = Encoding.UTF8.GetBytes(@"[");
+                    byte[] endArray = Encoding.UTF8.GetBytes(@"]");
+
+                    var stream = new MemoryStream();
+
+                    stream.Write(utf8Bom, 0, utf8Bom.Length);
+                    stream.Write(startArray, 0, startArray.Length);
+
+                    for (int i = 1; i <= count; i++)
+                    {
+                        stream.Write(objBytes, 0, objBytes.Length);
+
+                        if (i < count)
+                        {
+                            stream.Write(comma, 0, comma.Length);
```

Comment:
nit: For these single byte values (like comma, startArray, endArray), why not just use the WriteByte API?

Reply:
Sure. Thanks